scripts/tasks/download-translations.js

```diff
@@ -0,0 +1,170 @@
+
+const { translationsApi } = new crowdin({
+  token: CROWDIN_TOKEN,
+});
+
+/**
+ * Downloads the Crowdin file and unzips the contents
+ * @param {string} url
+ * @param {string} destination
+ */
+const downloadFiles = async (url, destination) => {
+  const tmpPath = join(
+    tmpdir(),
+    'electronjs.org',
+    `${Math.ceil(Math.random() * 1000)}`
+  );
+
+  await pipeline(got.stream(url), unzipper.Extract({ path: tmpPath }));
+
+  const contentPath = join(tmpPath, `[electron.i18n] master`, `content`);
+
+  const locales = await fs.readdir(contentPath);
+
+  for (const locale of locales) {
+    const localeDestination = join(destination, locale.toLowerCase());
+    const docsDestination = join(
+      localeDestination,
+      'docusaurus-plugin-content-docs',
+      'current',
+      'latest'
+    );
+    await fs.mkdirp(docsDestination);
+
+    await fs.copy(join(contentPath, locale, 'docs'), docsDestination);
+
+    await fs.copy(
+      join(contentPath, locale, 'website', 'i18n'),
+
+      localeDestination
+    );
+  }
+
+  return locales.map((locale) => locale.toLowerCase());
+};
+
+/**
+ * Waits for the given number of seconds
+ * @param {number} seconds
+ * @returns
+ */
+const waitFor = (seconds) => {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve();
+    }, seconds * 1000);
+  });
+};
+
+/**
+ * @param {number} buildId
+ * @returns
+ */
+const getBuild = async (buildId) => {
+  const builds = await translationsApi.listProjectBuilds(PROJECT_ID);
+  const build = builds.data.find((item) => item.data.id === buildId);
+
+  return build.data;
+};
+
+/**
+ * Kicks a build for `PROJECT_ID` and returns the download link once it
+ * has finished.
+ * If a build is not kicked we risked downloading outdated files.
+ * @param {number} projectId
+ */
+const buildAndDownloadLink = async (projectId) => {
+  const {
+    data: { id: buildId },
+  } = await translationsApi.buildProject(projectId);
+
+  let counter = 10;
+  let interval = 30;
+  let build;
+
+  for (let i = 0; i < counter; i++) {
+    build = await getBuild(buildId);
+    console.log(`Crowdin status: Project ${buildId} - ${build.status}`);
+
+    if (build.status === 'finished') {
+      break;
+    } else {
+      console.log(
+        `Crowdin status: Waiting ${interval} seconds (retry ${i}/${counter})`
+      );
+      await waitFor(interval);
+    }
+  }
+
+  if (build.status !== 'finished') {
+    throw new Error(`The project didn't build fast enough on Crowdin`);
+  }
+
+  const {
+    data: { url },
+  } = await translationsApi.downloadTranslations(PROJECT_ID, buildId);
+
+  return url;
+};
+
+/**
+ * Gets the download link for the latest available build
+ * @param {number} projectId
+ */
+const getLatestBuildLink = async (projectId) => {
+  const { data: builds } = await translationsApi.listProjectBuilds(projectId);
+
+  const {
+    data: { url },
+  } = await translationsApi.downloadTranslations(PROJECT_ID, builds[0].data.id);
```

Comment:
This assumes that the build at the front of the array is always the latest available. That may well be the case but I don't see that speced anywhere in the in crowdin docs. If you're sure this is correct, carry on :smile:

Reply:
For some reason I only get an array of 1 item no matter how many times I call it or kick a build. I will add a comment here explaining that.